src/plugins/content_management/public/components/section_input.ts

```diff
@@ -0,0 +1,196 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { BehaviorSubject } from 'rxjs';
+import { SavedObjectsClientContract } from 'opensearch-dashboards/public';
+
+import { Content, Section } from '../services';
+import { ViewMode } from '../../../embeddable/public';
+import { DashboardContainerInput, SavedObjectDashboard } from '../../../dashboard/public';
+import { CUSTOM_CONTENT_EMBEDDABLE } from './custom_content_embeddable';
+import { CardContainerInput } from './card_container/card_container';
+import { CARD_EMBEDDABLE } from './card_container/card_embeddable';
+
+const DASHBOARD_GRID_COLUMN_COUNT = 48;
+
+export const createCardInput = (
+  section: Section,
+  contents: Content[]
+): CardContainerInput | null => {
+  if (section.kind !== 'card') {
+    throw new Error(`function does not support section.type: ${section.kind}`);
+  }
+
+  const panels: CardContainerInput['panels'] = {};
+
+  const input: CardContainerInput = {
+    id: section.id,
+    title: section.title ?? '',
+    hidePanelTitles: true,
+    viewMode: ViewMode.VIEW,
+    panels,
+  };
+
+  contents.forEach((content) => {
+    if (content.kind === 'card') {
+      panels[content.id] = {
+        type: CARD_EMBEDDABLE,
+        explicitInput: {
+          id: content.id,
+          title: content.title,
+          description: content.description,
+          onClick: content.onClick,
+        },
+      };
+    }
+  });
+
+  if (Object.keys(panels).length === 0) {
+    return null;
+  }
+
+  return input;
+};
+
+export const createDashboardInput = async (
```

Comment:
It is actually the dashboard container, it uses the same embeddable factory as dashboard container use, you can find it in `section_render.tsx`
```typescript
  const factory = embeddable.getEmbeddableFactory('dashboard');
```

Here I'm creating dashboard inputs myself is because I need to not only support rendering content directly with a dashboard saved object, but also I need to render with contents that user added to the dashboard section, such as individual visualizations, or arbitrary embeddables, so I need to compute the layout on the fly